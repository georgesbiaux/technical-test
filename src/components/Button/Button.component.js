import React from 'react';
import StyledButton from './Button.style';

const Button = props => (
  <StyledButton
    className="priority-button"
    priority={props.priority}
    onClick={props.onClick}
  >
    <i className={`fa ${props.icon}`} />
  </StyledButton>
);

export default Button;
