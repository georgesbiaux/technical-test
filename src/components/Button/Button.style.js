import styled from 'styled-components';

const StyledButton = styled.button`
  border: 0;
  font-size: 20px;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  background-color: ${props => (props.priority ? '#75d286' : '#343434')};
  color: #ffffff;
  cursor: pointer;
  margin: 5px;
  transition: background-color 0.2s ease;

  :focus {
    outline: -webkit-focus-ring-color auto 0px;
  }

  :hover {
    background-color: ${props => (props.priority ? '#599e65' : '#343434')};
  }
`;

export default StyledButton;
