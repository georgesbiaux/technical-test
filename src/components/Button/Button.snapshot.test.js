import React from 'react';
import { render } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Button from './Button.component';

describe('<Button>', () => {
  it('should render correctly when priority is false', () => {
    const props = {
      priority: false,
      icon: 'fa-bars',
    };
    const tree = render(<Button {...props} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });

  it('should render correctly when priority is true', () => {
    const props = {
      priority: true,
      icon: 'fa-bars',
    };
    const tree = render(<Button {...props} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
