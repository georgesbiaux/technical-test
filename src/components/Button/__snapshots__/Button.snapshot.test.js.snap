// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> should render correctly when priority is false 1`] = `
<button
  class="priority-button sc-bdVaJa kWSmXR"
>
  <i
    class="fa fa-bars"
  >
    
  </i>
</button>
`;

exports[`<Button> should render correctly when priority is true 1`] = `
<button
  class="priority-button sc-bdVaJa brwQhq"
>
  <i
    class="fa fa-bars"
  >
    
  </i>
</button>
`;
